sql/core/src/main/scala/org/apache/spark/sql/execution/SparkSqlParser.scala

```diff
@@ -713,13 +713,40 @@ class SparkSqlAstBuilder(conf: SQLConf) extends AstBuilder {
   }
 
   /**
-   * Create an [[AddJar]] or [[AddFile]] command depending on the requested resource.
+   * A resource command depending on the requested operation on resources.
+   * Expected format:
+   * {{{
+   *   ADD (FILE[s] [filepath ...] | JAR[s] [jarpath ...])
+   *   DELETE (FILE[s] [filepath ...] | JAR[s] [jarpath ...])
+   *   LIST (FILE[s] [filepath ...] | JAR[s] [jarpath ...])
+   * }}}
    */
-  override def visitAddResource(ctx: AddResourceContext): LogicalPlan = withOrigin(ctx) {
-    ctx.identifier.getText.toLowerCase match {
-      case "file" => AddFile(remainder(ctx.identifier).trim)
-      case "jar" => AddJar(remainder(ctx.identifier).trim)
-      case other => throw operationNotAllowed(s"ADD with resource type '$other'", ctx)
+  override def visitManageResource(ctx: ManageResourceContext): LogicalPlan = withOrigin(ctx) {
+    ctx.op.getType match {
+      case SqlBaseParser.ADD =>
+        ctx.identifier.getText.toLowerCase match {
+          case "file" => AddFile(remainder(ctx.identifier).trim)
+          case "jar" => AddJar(remainder(ctx.identifier).trim)
+          case other => throw operationNotAllowed(s"ADD with resource type '$other'", ctx)
+        }
+      case SqlBaseParser.DELETE =>
+        throw operationNotAllowed(s"DELETE resources", ctx)
+      case SqlBaseParser.LIST =>
+        ctx.identifier.getText.toLowerCase match {
+          case "files" | "file" =>
+            if (remainder(ctx.identifier).trim.length > 0) {
```

Comment:
nit: we can define a variable `val path = remainder(ctx.identifier).trim` at the very beginning.